JunList.java

```diff
@@ -19,99 +21,104 @@ public void append(int value) {
             int[] newList = new int[list.length * 2];
             for (int i = 0; i < list.length; i++) {
                 newList[i] = this.list[i];
-                this.list = newList;
             }
-            this.list[trackListPosition] = value;
-            trackListPosition += 1;
+            this.list = newList;
         }
+        this.list[trackListPosition] = value;
+        trackListPosition += 1;
+    }
 
-        public void add(int position,int value){
-            //creates list twice as big when necessary
-            if (trackListPosition == list.length) {
-                int[] newList = new int[list.length * 2];
-                for (int i = 0; i < list.length; ++i) {
-                    if (i < position) {
-                        newList[i] = list[i];
-                    } else if (position == i) {
-                        newList[i] = value;
-                    } else {
-                        newList[i] = list[i - 1];
-                    }
-                }
-                this.list = newList;
-            } else {
-                for (int i = trackListPosition; i > position; i--) {
-                    list[i] = list[i - 1];
+    public int get(int index) {
+        return this.list[index];
+    }
+
+    public void add(int position, int value) {
+        //creates list twice as big when necessary
+        if (trackListPosition == list.length) {
+            int[] newList = new int[list.length * 2];
+            for (int i = 0; i < list.length; ++i) {
+                if (i < position) {
+                    newList[i] = list[i];

```

Comment:
Whenever I'm reading your code, I'm normally high af at this point in the night, tbh even if your logic in this method was broken I'd have no idea. So on your remove and add methods, just make sure you test them deeply because I literally never check your logic.